internal/provider/resources/work_pool.go

```diff
@@ -434,3 +401,31 @@ func (r *WorkPoolResource) ImportState(ctx context.Context, req resource.ImportS
 		resource.ImportStatePassthroughID(ctx, path.Root("name"), req, resp)
 	}
 }
+
+// BaseJobTemplatesMatch checks to see if two base_job_template objects are
+// equivalent, accounting for differences in the order of the contents.
+//
+//nolint:ireturn // required by Terraform API
+func BaseJobTemplatesMatch(bjtPlan, bjtReceived map[string]interface{}) diag.Diagnostic {
```

Comment:
- This function is exported because I needed to use it in the work pool tests to tell it how to accurately compare the fields.
- This got me thinking that it may be worth making this function more generic (i.e. not specific to base job templates) so we can use it to compare objects elsewhere in the provider. Open to doing that now or in a future PR (if we keep this setup, see earlier comment).

Reply:
With `go-test/deep`, the function was flexible enough that I moved it into the helpers package.